core/app/models/spree/shipping_rate.rb

```diff
@@ -1,46 +1,48 @@
 module Spree
   class ShippingRate < Spree::Base
+    attr_writer :order
```

Comment:
Is this here primarily because the Coordinator builds shipping rates before it builds shipments?  Are there other places where we need to build shipping rates without a shipment in hand?  Does it make sense to have a shipping rate without a shipment? It just feels a bit awkward and I'm wondering if there's anything reasonable we could do to avoid this.